Couenne options while using AmplNLWriter

Hi,
I am using Couenne solver via AmplNLWriter.jl in Julia. 
I created an option file **couenne.opt**: 
{
allowable_fraction_gap 0.05
time_limit 3600
}
For an instance of my problem I am getting solutions however for another instance, I am getting an AssertionError. I couldn`t understand the cause of the problem and how to solve it. Below you can find the error log regarding the issue. I will be glad if you can help me regarding this or share a source to reach.

Regards,
Berkay GULCAN

**Error log:** 
{
ERROR: LoadError: AssertionError: line[1:7] == "Options"
Stacktrace:
 [1] read_sol(::IOStream, ::AmplNLWriter.AmplNLMathProgModel) at /home/bgulcan/.julia/v0.6/AmplNLWriter/src/AmplNLWriter.jl:666
 [2] read_results(::IOStream, ::AmplNLWriter.AmplNLMathProgModel) at /home/bgulcan/.julia/v0.6/AmplNLWriter/src/AmplNLWriter.jl:575
 [3] open(::AmplNLWriter.##117#118{AmplNLWriter.AmplNLMathProgModel}, ::String, ::String) at ./iostream.jl:152
 [4] read_results(::AmplNLWriter.AmplNLMathProgModel) at /home/bgulcan/.julia/v0.6/AmplNLWriter/src/AmplNLWriter.jl:569
 [5] optimize!(::AmplNLWriter.AmplNLMathProgModel) at /home/bgulcan/.julia/v0.6/AmplNLWriter/src/AmplNLWriter.jl:419
 [6] optimize!(::AmplNLWriter.AmplNLNonlinearModel) at /home/bgulcan/.julia/v0.6/AmplNLWriter/src/AmplNLWriter.jl:772
 [7] #solvenlp#165(::Bool, ::Function, ::JuMP.Model, ::JuMP.ProblemTraits) at /home/bgulcan/.julia/v0.6/JuMP/src/nlp.jl:1271
 [8] (::JuMP.#kw##solvenlp)(::Array{Any,1}, ::JuMP.#solvenlp, ::JuMP.Model, ::JuMP.ProblemTraits) at ./<missing>:0
 [9] #solve#116(::Bool, ::Bool, ::Bool, ::Array{Any,1}, ::Function, ::JuMP.Model) at /home/bgulcan/.julia/v0.6/JuMP/src/solvers.jl:172
 [10] createandsolvemodel_DCT(::Biomass, ::Int64, ::Int64, ::Int64, ::Int64, ::Float64, ::Float64, ::Float64, ::Array{Float64,3}, ::Array{Float64,3}, ::Array{Float64,1}, ::Array{Float64,2}, ::Int64, ::Float64, ::Array{Float64,3}, ::Array{Float64,3}, ::Array{Float64,1}) at /home/bgulcan/Model-SmallestFull-Couenne3.jl:139
 [11] macro expansion at /home/bgulcan/Run-SmallestFull-Couenne3.jl:115 [inlined]
 [12] anonymous at ./<missing>:?
 [13] include_from_node1(::String) at ./loading.jl:576
 [14] include(::String) at ./sysimg.jl:14
 [15] process_options(::Base.JLOptions) at ./client.jl:305
 [16] _start() at ./client.jl:371
while loading /home/bgulcan/Run-SmallestFull-Couenne3.jl, in expression starting on line 113
}

--- a/examples/jump_minlp.jl
+++ b/examples/jump_minlp.jl
@@ -43,7 +43,7 @@ using JuMP, Base.Test, AmplNLWriter
 
     @test solve(m) == :Optimal
 
-    if getsolvername(solver) == "ipopt"
+    if contains(getsolvername(solver), "ipopt")
         # Ipopt solves the relaxation
         @test isapprox(getvalue(x)[:], [1.14652, 0.546596, 1.0], atol=1e-5)
         @test isapprox(getvalue(y)[:], [0.27330, 0.299959, 0.0], atol=1e-5)

--- a/examples/jump_nonlinearbinary.jl
+++ b/examples/jump_nonlinearbinary.jl
@@ -23,7 +23,7 @@ using JuMP, Base.Test, AmplNLWriter
 
     @test solve(m) == :Optimal
 
-    if getsolvername(solver) == "ipopt"
+    if contains(getsolvername(solver), "ipopt")
         # Ipopt solves the relaxation
         @test isapprox(getvalue(x), [0.501245, 1.0], atol=1e-6)
         @test isapprox(getobjectivevalue(m), 0.249377, atol=1e-6)

--- a/src/AmplNLWriter.jl
+++ b/src/AmplNLWriter.jl
@@ -653,7 +653,7 @@ function read_sol(f::IO, m::AmplNLMathProgModel)
     # "Options".
     while true
         line = readline(f)
-        if line[1:7] == "Options"
+        if length(line) >= 7 && line[1:7] == "Options"
             break
         else
             m.solve_message *= line

--- a/src/AmplNLWriter.jl
+++ b/src/AmplNLWriter.jl
@@ -394,7 +394,7 @@ function optimize!(m::AmplNLMathProgModel)
     end
     m.probfile = "$file_basepath.nl"
     m.solfile = "$file_basepath.sol"
-    
+
     try
         write_nl_file(f_prob, m)
     finally
@@ -649,17 +649,15 @@ function read_sol(f::IO, m::AmplNLMathProgModel)
     stat = :Undefined
     line = ""
 
-    # Keep building solver message by reading until first truly empty line
+    # Extract the solver message by concatenating all of the lines until
+    # "Options".
     while true
         line = readline(f)
-        isempty(chomp(line)) && break
-        m.solve_message *= line
-    end
-
-    # Skip over empty lines
-    while true
-        line = readline(f)
-        !isempty(chomp(line)) && break
+        if line[1:7] == "Options"
+            break
+        else
+            m.solve_message *= line
+        end
     end
 
     # Read through all the options. Direct copy of reference implementation.

--- a/test/runtests.jl
+++ b/test/runtests.jl
@@ -4,6 +4,7 @@ using Base.Test
 include("nl_convert.jl")
 include("nl_linearity.jl")
 include("nl_write.jl")
+include("sol_file_parser.jl")
 
 # needed for the scoping of `solver` in the examples
 solver = JuMP.UnsetSolver()

--- a/test/sol_file_parser.jl
+++ b/test/sol_file_parser.jl
@@ -1,0 +1,29 @@
+const FILE_80 = """
+
+ Couenne (/home/bgulcan/.julia/v0.6/AmplNLWriter/.solverdata/tmpMSxFvq.nl Mar  6 2018): Infeasible
+
+Options
+3
+0
+1
+0
+140
+0
+110
+0
+objno 0 220
+"""
+
+@testset "Test sol file parsing" begin
+    @testset "Issue #80" begin
+        io = IOBuffer()
+        write(io, FILE_80)
+        seekstart(io)
+        model = AmplNLWriter.AmplNLMathProgModel("", String[], "")
+        model.ncon = 140
+        model.nvar = 110
+        @test !AmplNLWriter.read_sol(io, model)
+        @test model.solve_result_num == 220
+        @test all(model.solution .=== NaN)
+    end
+end

